developer-docs-site/docs/move/book/variables.md

```diff
@@ -714,25 +714,30 @@ before it is assigned a value.
 As mentioned above, the Move compiler will infer a `copy` or `move` if one is not indicated. The
 algorithm for doing so is quite simple:
 
-- Any scalar value with the `copy` [ability](./abilities.md) is given a `copy`.
+- Any value with the `copy` [ability](./abilities.md) is given a `copy`.
 - Any reference (both mutable `&mut` and immutable `&`) is given a `copy`.
   - Except under special circumstances where it is made a `move` for predictable borrow checker
     errors.
 - Any other value is given a `move`.
```

Comment:
If the compiler can prove that the source value with `copy` ability is not used after the assignment, then a move may be used instead of a copy for performance, but this will be invisible to the programmer (except in possible decreased time or gas cost).

Reply:
done

Reply:
BTW, maybe the following code demonstrates the compiler will always prefer `move` if possible.

```move
    struct Car has store, copy {}
    struct User has key {
        car: Car
    }
    #[test]
    fun test_move_and_copy() {
        let s1 = create_signer_for_test(@0xcafe);
        let s2 = create_signer_for_test(@0xabcd);

        let c = Car {};
        let u1 = User { car: copy c }; // copy
        let u2 = User { car: move c }; // move, since car can not be dropped, and this test passed.
        move_to(&s1, u1);
        move_to(&s2, u2);
    }
```